File: src/Stronghold.Application/Features/Appointments/AppointmentResponse.cs
namespace Stronghold.Application.Features.Appointments;

public class AppointmentResponse
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string UserName { get; set; } = string.Empty;
    public int StaffId { get; set; }
    public string StaffName { get; set; } = string.Empty;
    public string StaffType { get; set; } = string.Empty;
    public DateTime ScheduledAt { get; set; }
    public int DurationMinutes { get; set; }
    public string Status { get; set; } = string.Empty;
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: stronghold_desktop/lib/features/staff/models/appointment_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'appointment_response.freezed.dart';
part 'appointment_response.g.dart';

@freezed
abstract class AppointmentResponse with _$AppointmentResponse {
  const factory AppointmentResponse({
    required int id,
    required int userId,
    required String userName,
    required int staffId,
    required String staffName,
    required String staffType,
    required DateTime scheduledAt,
    required int durationMinutes,
    required String status,
    String? notes,
    required DateTime createdAt,
  }) = _AppointmentResponse;

  factory AppointmentResponse.fromJson(Map<String, dynamic> json) =>
      _$AppointmentResponseFromJson(json);
}

@freezed
abstract class PagedAppointmentResponse with _$PagedAppointmentResponse {
  const factory PagedAppointmentResponse({
    required List<AppointmentResponse> items,
    required int totalCount,
    required int totalPages,
    required int currentPage,
    required int pageSize,
  }) = _PagedAppointmentResponse;

  factory PagedAppointmentResponse.fromJson(Map<String, dynamic> json) =>
      _$PagedAppointmentResponseFromJson(json);
}
